lib/index.js

```diff
@@ -12,13 +12,15 @@ function AuthPassportLocal () {
   this.context = {};
   this.pipes = pipes;
 
-  this.init = function (config, context, isDummy) {
+  this.init = function (customConfig, context, isDummy) {
     var strategy;
+    const defaultConfig = {
+      'secret': 'changeme',
```

Comment:
@scottinet I agree with you on the fact that machines should generate the salt. I also agree that, asking the user for salt is not a common (nor really a great) user experience.
The only motivation to do this is related to the clustering problem, but again, this means delegating _two_ responsibilities to the user: generating the salt _and_ sharing it among cluster nodes.

My last thought is that I agree with you, ~Scott~ Seb, and that we should find a way to make this transparent to the user.

Now, foolproofing is the issue. Not sure if it's a good idea but... What about storing the generated salt into the Elasticsearch reserved index and falling back to a default salt if not present?
Also, in the case of a shared, generated salt, which node would  be responsible for the generation? The master?